TypeCobol/AntlrGrammar/CobolCodeElements.g4

```diff
@@ -307,7 +307,7 @@ cobol2002Statement: UNSAFE;
 programIdentification:
 	(IDENTIFICATION | ID) DIVISION PeriodSeparator 
 	PROGRAM_ID PeriodSeparator? programNameDefinition
-	(IS? (RECURSIVE | INITIAL | (COMMON INITIAL?) | (INITIAL COMMON?)) PROGRAM?)? PeriodSeparator?
+	(IS? (RECURSIVE | INITIAL | (COMMON INITIAL?) | (INITIAL COMMON?)) PROGRAM?)? PeriodSeparator
```

Comment:
The Cobol ISO 2014 specifications states that PeriodSeparator is mandatory but the IBM Enterprise Cobol 6.1 states that the PeriodSeparator is optional.

Some of our production programs don't have a PeriodSeparator.
So I think our grammar must still allow PeriodSeparator to be optional.

We can generate a warning if this PeriodSeparator is missing.

What do you think?

Reply:
I'm not sure, the underlying question is whether we follow Cobol ISO 2014 or IBM Enterprise Cobol 6.1?

Reply:
add warnings when missing dot after PROGRAM-ID or PROGRAM-ID declaration (Cobol ISO 2014)
https://github.com/TypeCobolTeam/TypeCobol/pull/1651/commits/fa8312e26b5068bc733e0bd9c28c80005bd568dd

Reply:
We mainly target IBM Enterprise Cobol 6.1, but we should also try to keep compatibility with others Cobol if possible.